design/ceph/ceph-orchestrator-osd-add-remove.md

```diff
@@ -0,0 +1,266 @@
+it permission to edit and remove the item in the `zapOSDs` list. Users can be made aware that if
+they set this value `true`, the Orchestrator Module may modify or delete that item automatically.
+
+When the Orchestrator Module specifies an OSD to remove in the `CephCluster` resource, the Module
+should also edit the Drive Group(s) add the corresponding disk to the blacklist section of the Drive
+Group(s) (feature request #2) if the user wishes to zap the disk so that it won't be unintentionally
+automatically added back into the cluster as an OSD by Rook via Drive Groups (though other `storage`
+spec methods might still add the disk back). The user can later remove blacklisted disks if they so
+desire.
+
+We should document for the user that OSDs should be removed before being zapped; however, it is not
+critical to enforce this, as a zapped OSD will eventually be marked as `out` and `safe-to-remove`.
+The Orchestrator Module, being a good citizen, should perform steps in the correct order to remove
+the OSD completely before zapping it.
+
+#### Reporting status
+Rook should report OSD zapping success or failure in a `status` field.
+
+```yaml
+apiVersion: ceph.rook.io/v1
+kind: CephCluster
+status:
+  zappedOSDs: # proposed new
+    - FSID: < string, FSID of OSD to remove > # proposed new
+      successfullyZapped: < bool, true if removed, false if not >
+      message: < string, message from results of removal operation >
+```
+
+There should be an `zappedOSDs` status entry for each entry in the `zapOSDs` list from the
+`storage` spec. Status entries should be removed when the `zapOSDs` spec no longer contains them,
+and new status entries should be added when `zapOSDs` items are added.
+
+If the OSD has not yet been zapped (a status entry for a newly-added `zapOSDs` item), the entry
+should report `successfullyZapped: false` with a message indicating Rook has not yet attempted to
+zap the device.
+
+If all OSD provisioning Jobs run without finding an OSD matching the FSID, the entry should report
+`successfullyZapped: true` with a message indicating the OSD with the given FSID already does not
+exist.
+
+Rook's OSD provisioning Job should not attempt to search for or zap OSDs which already have a
+`successfullyZapped: true` status so as to (1) not waste time searching for an already zapped
+device, and (2) not overwrite the initial success message from the OSD being zapped.
+
+If the `controlledByCephOrchestrator: true` field is set in the `spec`, the Rook Orchestrator Module
+should remove items from `zapOSDs` in the `CephCluster` resource when the FSID is reported as
+successfully zapped so as to keep the `zapOSDs` spec and `zappedOSDs` status items short. CRD users
+should be advised to remove items from `zapOSDs` when the OSD is reported as `successfullyZapped`
+for the same reason or to set `controlledByCephOrchestrator: true` in the spec.
+
```

Comment:
I think the same that i have explained for having a section to descibe the "OSD creation" status.  This is an operation that probably has more sense to check in logs.
Apart of that, i think that we are starting to use the CRD as a box where almost everything can be stored. I thing that to keep in the CRD only the declarative description of the cluster probably is the right path to follow.